src/umpire/resource/HipDeviceMemoryResource.cpp

```diff
@@ -65,6 +65,30 @@ std::size_t HipDeviceMemoryResource::getHighWatermark() const noexcept
   return 0;
 }
 
+bool HipDeviceMemoryResource::isHostPageable() noexcept
+{
+  hipDeviceProp_t props;
+  int hdev = m_traits.id;
+  hipGetDevice(&hdev);
+
+  //Check whether HIP can map host memory.
+  hipGetDeviceProperties(&props, hdev);
+  if(props.canMapHostMemory)
+    return true;
+  else
+    return false;
+}
+
+bool HipDeviceMemoryResource::isAccessibleFrom(Platform p) noexcept
+{
+  if(p == Platform::hip)
+    return true;
+  else if(p == Platform::host)
+    return isHostPageable();
```

Comment:
I don't think you need to check this. As far as I know, HIP device memory will always be accessible from the host.

Reply:
I updated this check so that if its the host or hip it will return true, otherwise return false